packages/vertexai/lib/methods/generate-content.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  GenerateContentRequest,
+  GenerateContentResponse,
+  GenerateContentResult,
+  GenerateContentStreamResult,
+  RequestOptions,
+} from '../types';
+import { Task, makeRequest } from '../requests/request';
+import { createEnhancedContentResponse } from '../requests/response-helpers';
+import { processStream } from '../requests/stream-reader';
+import { ApiSettings } from '../types/internal';
+
+export async function generateContentStream(
+  apiSettings: ApiSettings,
+  model: string,
+  params: GenerateContentRequest,
+  requestOptions?: RequestOptions,
+): Promise<GenerateContentStreamResult> {
+  const response = await makeRequest(
+    model,
+    Task.STREAM_GENERATE_CONTENT,
+    apiSettings,
+    /* stream */ true,
+    JSON.stringify(params),
+    requestOptions,
+  );
+  return processStream(response);
+}
+
+export async function generateContent(
+  apiSettings: ApiSettings,
+  model: string,
+  params: GenerateContentRequest,
+  requestOptions?: RequestOptions,
+): Promise<GenerateContentResult> {
+  const response = await makeRequest(
+    model,
+    Task.GENERATE_CONTENT,
+    apiSettings,
+    /* stream */ false,
```

Comment:
after all the streaming polyfill part !? stream false 😆 - honestly though I'm not sure what this actually does, but it caught my eye

Reply:
ha, there's essentially 2 ways to generate content. Standard HTTP request and a stream request. The function above this is the stream request: https://github.com/invertase/react-native-firebase/blob/vertexai/packages/vertexai/lib/methods/generate-content.ts#L30-L45